server/src/main/java/io/spine/system/server/DefaultSystemGateway.java

```diff
@@ -79,6 +83,21 @@ public void postEvent(Message systemEvent) {
               .post(event, noOpObserver());
     }
 
+    @Override
+    public Iterator<Any> readDomainAggregate(Query query) {
+        @SuppressWarnings("unchecked") // Logically checked.
+        MirrorRepository repository = (MirrorRepository)
+                system.findRepository(Mirror.class)
+                      .orElseThrow(
+                              () -> newIllegalStateException(
```

Comment:
Let's discuss (in person!) if this is a proper behavior in a context of propagating this exception to the client.